api/route/pokemons.route.spec.ts

```diff
@@ -184,3 +185,834 @@ describe("GET /pokemons/random", () => {
+          speed: 72,
+          description: "A normal Pokémon",
+          image: "image_url",
+          type: ["Normal"],
+          abilities: ["Run Away"],
+          height: "0.3 m",
+          weight: "3.5 kg",
+        },
+        {
+          id: "7",
+          name: "Clefairy",
+          hp: 70,
+          attack: 45,
+          defense: 40,
+          speed: 56,
+          description: "A flying Pokémon",
+          image: "image_url",
+          type: ["Normal", "Flying"],
+          abilities: ["Keen Eye"],
+          height: "0.3 m",
+          weight: "1.8 kg",
+        },
+        {
+          id: "8",
+          name: "Cobalion",
+          hp: 115,
+          attack: 45,
+          defense: 20,
+          speed: 20,
+          description: "A fairy Pokémon",
+          image: "image_url",
+          type: ["Fairy"],
+          abilities: ["Cute Charm"],
+          height: "0.5 m",
+          weight: "5.5 kg",
+        },
+        {
+          id: "9",
+          name: "Corphish",
+          hp: 80,
+          attack: 33,
+          defense: 35,
+          speed: 55,
+          description: "A poison Pokémon",
+          image: "image_url",
+          type: ["Poison", "Flying"],
+          abilities: ["Inner Focus"],
+          height: "0.8 m",
+          weight: "7.5 kg",
+        },
+        {
+          id: "10",
+          name: "Meowth",
+          hp: 40,
+          attack: 45,
+          defense: 35,
+          speed: 90,
+          description: "A normal Pokémon",
+          image: "image_url",
+          type: ["Normal"],
+          abilities: ["Pickup"],
+          height: "0.4 m",
+          weight: "4.2 kg",
+        },
+      ],
+    });
+
+    await prisma.userPokemon.createMany({
+      data: [
+        { user_id: userId, pokemon_id: "1" },
+        { user_id: userId, pokemon_id: "2" },
+        { user_id: userId, pokemon_id: "4" },
+        { user_id: userId, pokemon_id: "6" },
+        { user_id: userId, pokemon_id: "7" },
+        { user_id: userId, pokemon_id: "8" },
+      ],
+    });
+
+    const response = await request(app)
+      .post("/pokemons?page=1&limit=3&sort=HP (High to low)")
+      .send({ name: "C", user_id: userId })
+      .expect(200);
+
+    expect(response.body).toHaveLength(3);
+    expect(response.body[0].name).toBe("Cobalion");
+    expect(response.body[0].hp).toBe(115);
+
+    expect(response.body[1].name).toBe("Claydol");
+    expect(response.body[1].hp).toBe(100);
+  });
+
+  it("should return an empty array if there are no Pokémon in the table", async () => {
+    const response = await request(app)
+      .post(`/pokemons`)
+      .send({ name: "CCC" })
+      .expect(200);
+
+    expect(response.body).toEqual([]);
+  });
+
+  it("should return an empty array for a user with no owned pokemons", async () => {
+    const response = await request(app)
+      .post("/pokemons")
+      .send({ user_id: userId })
+      .expect(200);
+
+    expect(response.body).toEqual([]);
```

Comment:
i think you should add pokemons to the db or to another user and then check if the response is an empty array.
if there are no pokemons in the db the response will be an empty array anyway so your test is not checking this case